netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -150,7 +152,9 @@ static NettyClientHandler newHandler(
     Preconditions.checkArgument(maxHeaderListSize > 0, "maxHeaderListSize must be positive");
     Http2HeadersDecoder headersDecoder = new GrpcHttp2ClientHeadersDecoder(maxHeaderListSize);
     Http2FrameReader frameReader = new DefaultHttp2FrameReader(headersDecoder);
-    Http2FrameWriter frameWriter = new DefaultHttp2FrameWriter();
+    Http2HeadersEncoder encoder = new DefaultHttp2HeadersEncoder(
```

Comment:
How does this not get thrown away when you call newHandler below?

Reply:
It is passed to the frameWriter on the next line.